gipc/gipc.py

```diff
@@ -1058,6 +1061,33 @@ def put(self, o):
             self._write(struct.pack("!i", len(bindata)) + bindata)
 
 
+if not WINDOWS and FORK_MODE == 'spawn':
+    def reduce_GIPCReader(reader):
+        df = multiprocessing.reduction.DupFd(reader._fd)
+        return (rebuild_GIPCReader, (df, reader._decoder))
+
+
+    def rebuild_GIPCReader(df, decoder):
+        fd = df.detach()
```

Comment:
For my understanding, sorry for misusing your time here -- what is `detach()` doing?

Reply:
I don't know, but it is what reduction does after forking: https://github.com/python/cpython/blob/af5fb6706219d7949c1db5c9f2b7da53198123f3/Lib/multiprocessing/reduction.py#L242-L248

Seems like it's sending the handle over a socket and detach receives the handle
https://github.com/python/cpython/blob/af5fb6706219d7949c1db5c9f2b7da53198123f3/Lib/multiprocessing/resource_sharer.py#L45-L58